### PR TITLE
Fw hierarchy changes

### PR DESCRIPTION
**Pull request by krgauraw**

### CODE REVIEW

File: platform-modules/actors/src/main/java/org/ekstep/learning/framework/FrameworkStore.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import java.util.List;
+import java.util.Map;
+
+/**
+ * FrameworkStore handles all CRUD Operation into cassandra for Framework Hierarchy.
+ * @author Kumar Gauraw
+ */
+public class FrameworkStore extends CassandraStore {
+
+    private ObjectMapper mapper = new ObjectMapper();
```
Comment (maheshkumargangula) [inline platform-modules/actors/src/main/java/org/ekstep/learning/framework/FrameworkStore.java]: Move it (ObjectMapper) to BaseClass and make it protected.

File: platform-modules/actors/src/main/java/org/ekstep/learning/framework/FrameworkStore.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
+package org.ekstep.learning.framework;
```
Comment (maheshkumargangula) [inline platform-modules/actors/src/main/java/org/ekstep/learning/framework/FrameworkStore.java]: This class is almost same as **CollectionStore** class. Why don't you reuse the same Class.

File: platform-modules/ekstep-framework/src/main/java/org/ekstep/framework/mgr/impl/FrameworkManagerImpl.java [hunk 3]
```diff
@@ -104,12 +89,14 @@ public Response createFramework(Map<String, Object> request, String channelId) t
 	@Override
 	public Response readFramework(String frameworkId, List<String> returnCategories) throws Exception {
 		Response response = new Response();
-		if (Platform.config.hasPath("framework.es.sync")) {
-			if (Platform.config.getBoolean("framework.es.sync")) {
+		if (Platform.config.hasPath("framework.cassandra.sync")) {
```
Comment (maheshkumargangula) [inline platform-modules/ekstep-framework/src/main/java/org/ekstep/framework/mgr/impl/FrameworkManagerImpl.java]: Remove this configuration (framework.cassandra.sync). We should always push the data to cassandra when it is published.

File: platform-modules/actors/src/main/java/org/ekstep/learning/framework/FrameworkStore.java (hunk 1 above)
Comment (krgauraw) [reply platform-modules/actors/src/main/java/org/ekstep/learning/framework/FrameworkStore.java]: done

File: platform-modules/actors/src/main/java/org/ekstep/learning/framework/FrameworkStore.java (hunk 2 above)
Comment (krgauraw) [reply platform-modules/actors/src/main/java/org/ekstep/learning/framework/FrameworkStore.java]: Deleted FrameworkStore and Renamed CollectionStore to HierarchyStore and used for framework.

File: platform-modules/ekstep-framework/src/main/java/org/ekstep/framework/mgr/impl/FrameworkManagerImpl.java (hunk 3 above)
Comment (krgauraw) [reply platform-modules/ekstep-framework/src/main/java/org/ekstep/framework/mgr/impl/FrameworkManagerImpl.java]: This property was used in Read Framework API Only. Removed. Now Read Framework API will try to fetch data from cassandra, if not present in cassandra, fetch it from neo4j.

File: platform-core/cassandra-connector/src/main/java/org/ekstep/cassandra/store/CassandraStore.java [hunk 4]
```diff
@@ -32,6 +33,8 @@
  */
 public abstract class CassandraStore {
 
+	protected ObjectMapper mapper = new ObjectMapper();
```
Comment (amitpriyadarshi) [inline platform-core/cassandra-connector/src/main/java/org/ekstep/cassandra/store/CassandraStore.java]: Why we have added this declaration. Not used any where.

File: platform-modules/ekstep-framework/src/main/java/org/ekstep/framework/mgr/impl/FrameworkManagerImpl.java [hunk 5]
```diff
@@ -84,61 +69,44 @@ public Response createFramework(Map<String, Object> request, String channelId) t
...
-								} else {
-									responseMap.put("categories", categories);
-								}
-							}
+		Map<String, Object> responseMap = new HashMap<>();
+		Response getHierarchyResp = getFrameworkHierarchy(frameworkId);
+		Map<String, Object> framework = (Map<String, Object>) getHierarchyResp.get("framework");
+
+		if (MapUtils.isNotEmpty(framework)) {
+			if (null != framework.get("fw_hierarchy")) {
```
Comment (amitpriyadarshi) [inline platform-modules/ekstep-framework/src/main/java/org/ekstep/framework/mgr/impl/FrameworkManagerImpl.java]: @krgauraw Do you feel, still we should use the same logic.. We can store and fetch directly complete framework hierarchy

Comment (krgauraw) [reply platform-modules/ekstep-framework/src/main/java/org/ekstep/framework/mgr/impl/FrameworkManagerImpl.java]: we are storing complete framework data not only hierarchy. So it is required as of now

File: platform-core/cassandra-connector/src/main/java/org/ekstep/cassandra/store/CassandraStore.java (hunk 4 above)
Comment (krgauraw) [reply platform-core/cassandra-connector/src/main/java/org/ekstep/cassandra/store/CassandraStore.java]: It is used in child class . HierarchyStore

**Pull request merged**